src/olympia/abuse/admin.py

```diff
@@ -185,6 +183,38 @@ def get_actions(self, request):
             actions.pop('mark_as_suspicious')
         return actions
 
+    def get_search_fields(self, request):
+        """
+        Return search fields according to the type filter.
+        """
+        type_ = request.GET.get('type')
+        if type_ == 'addon':
+            search_fields = (
+                'addon__name__localized_string', 'addon__slug', 'addon_name',
+                '=guid', 'message', 'addon__id',
+            )
+        elif type_ == 'user':
+            search_fields = (
+                'message', 'user__id', '^user__username', '^user__email',
+            )
+        else:
+            search_fields = ()
+        return search_fields
+
+    def get_search_id_field(self, request):
+        """
+        Return the field to use when all search terms are numeric, according to
+        the type filter.
+        """
+        type_ = request.GET.get('type')
+        if type_ == 'addon':
+            search_field = 'addon_id'
+        elif type_ == 'user':
+            search_field = 'user_id'
+        else:
+            search_field = 'pk'
```

Comment:
I don't suppose it matters, but `super().get_search_id_field(request)` instead when type isn't `addon` or `user`?

Reply:
it doesn't exist, it's a custom method

Reply:
You're overriding the mixin `get_search_id_field` method - why wouldn't it exist? https://github.com/mozilla/addons-server/pull/11438/files#diff-673ddf9dd55a8432ead7781c781b0d6fR10

Reply:
Ah, sorry, yeah, I can call `super()` and get the mixin implementation yes. Misread where the comment was.